### ISSUE TITLE
Project Comments

### ISSUE THREAD

**Issue by cycomachead** [bromagosa/snapCloud#108]
Users should be able to leave comments on a project, and should be able to reply to a comment.

Suggesting the following schema:

Comments Table
* user_id
* project_id
* parent_comment_id
* comment_text
* created_at
* updated_at

Maybe we should call `user_id` `author_id`...

**Comment by bromagosa** [comment]
Agreed, but as I said in some other issue, I don't think we should have comments for the PR.

Releasing the site will pour lots of issues on us by itself, so I'd rather wait until we size the amount of published projects and website users before adding a feature that needs moderation.

I think I recall Brian and Jens agreed we shouldn't have comments yet.

**Comment by cycomachead** [comment]
Ah, I had thought someone said we should have them, but I must have misread. Totally agree it’s ok to release the site without them.

**Comment by bromagosa** [comment]
I'm thinking we could leverage Discourse for these things. When someone writes a comment into a project page, a new thread could be automatically created in the forum under "Share your projects" where the project is displayed as the first post, and then comments are added as subsequent posts. To write new comments you'd be redirected to the discussion thread, but you could always see the whole comment thread from the project page itself.

Just thinking out loud and writing this down for the future. I'm not sure how easy does the Discourse API make this, but it seems redundant and time-consuming to create a commenting system when we have a forum that has a great one with flags, emojis and anything we could ever need :)

**Comment by cycomachead** [comment]
I've been thinking the same thing too. 

Apparently, a few people have tried similar things. I think it's worth a shot. Comments themselves aren't tricky, but definitely can be a time-suck.
